src/System.Private.CoreLib/src/System/Threading/WaitHandle.Windows.cs

```diff
@@ -168,7 +169,7 @@ private static bool WaitOneCore(IntPtr handle, int millisecondsTimeout)
             Debug.Assert(waitHandles.Length <= MaxWaitHandles);
             Debug.Assert(millisecondsTimeout >= -1);
 
-            int ret = WaitMultiple(currentThread, safeWaitHandles, waitHandles.Length, millisecondsTimeout, false /* waitany*/ );
+            int ret = WaitMultiple(currentThread, safeWaitHandles, numWaitHandles, millisecondsTimeout, false /* waitany*/ );
```

Comment:
Please see other uses of waitHandles.Length in this function, I believe all of them should be numWaitHandles instead